### PR TITLE
Exchange Issue into RB Set Using Eth

### PR DESCRIPTION
**Pull request by felix2feng**
ChangeLog:
- PayableExchangeIssue.sol is in the skipped files for coverage due to coverage's inability to instrument and handle .send/.transfer functions
- ICore: add interface for issueTo
- IExchangeIssueModule: expose exchange issue function
- ExchangeIssueLibrary: Export ExchangeIssue struct so it can be used elsewhere
- Create Mocks and interfaces for Wrapped Ether
- PayableExchangeIssue: Write functions for issueRebalancingSetWithEther
- Unit tests for payableExchangeIssue
- Supporting wrappers and web3 helper functions

### CODE REVIEW

File: contracts/supplementary/PayableExchangeIssue.sol [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    )
+        private
+        returns (uint256)
+    {
+        uint256[] memory rbSetUnits = ISetToken(_rebalancingSetAddress).getUnits();
+        uint256 rbSetUnit = rbSetUnits[0];
+        uint256 rbSetNaturalUnit = ISetToken(_rebalancingSetAddress).naturalUnit();
+
+        // Ensure that the base Set quantity is a multiple of the rebalancing Set natural unit
+        uint256 rbSetNormalizedBaseSetQuantity = _baseSetIssueQuantity.div(rbSetNaturalUnit).mul(rbSetNaturalUnit);
```
Comment (bweick) [inline contracts/supplementary/PayableExchangeIssue.sol]: I'm unsure what this line of code is supposed to do. Is this flipped with the line below? Wouldn't you want to see how many rebalancing set tokens you can issue and then make sure it's a multiple of the rebalancing set?

Comment (felix2feng) [reply contracts/supplementary/PayableExchangeIssue.sol]: +1 Think its flipped

File: contracts/core/lib/ExchangeIssueLibrary.sol [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * @title ExchangeIssueLibrary
+ * @author Set Protocol
+ *
+ * This library contains functions and structs to assist with parsing exchange issue data
+ */
+library ExchangeIssueLibrary {
+    // ============ Structs ============
+
+    struct ExchangeIssue {
```
[the remark this answers is not in the document]
Comment (felix2feng) [reply contracts/core/lib/ExchangeIssueLibrary.sol]: +1

File: contracts/supplementary/PayableExchangeIssue.sol [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        address _core,
+        address _transferProxy,
+        address _exchangeIssueModule,
+        address _wrappedEther
+    )
+        public
+    {
+        // Commit the address and instance of Core to state variables
+        core = _core;
+        coreInstance = ICore(_core);
```
[the remark this answers is not in the document]
Comment (felix2feng) [reply contracts/supplementary/PayableExchangeIssue.sol]: Slightly more gas efficient and cleans up multiple instantiations of instances

File: contracts/supplementary/PayableExchangeIssue.sol [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+     * @return rbSetIssueQuantity      The quantity of rebalancing Set to issue
+     */
+    function calculateRebalancingSetIssueQuantity(
+        address _rebalancingSetAddress,
+        uint256 _baseSetIssueQuantity
+    )
+        private
+        returns (uint256)
+    {
+        uint256[] memory rbSetUnits = ISetToken(_rebalancingSetAddress).getUnits();
```
[the remark this answers is not in the document]
Comment (felix2feng) [reply contracts/supplementary/PayableExchangeIssue.sol]: Hmmm good call. We have something called Unit shares already. I'll use that.

[the remark this answers is not in the document]
Comment (felix2feng) [reply contracts/supplementary/PayableExchangeIssue.sol]: +1

**Pull request merged**